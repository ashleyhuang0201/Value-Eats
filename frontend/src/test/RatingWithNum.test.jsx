import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import RatingWithNum from "../components/RatingWithNum";

describe("RatingWithNum", () => {
  it("can work in the base case", () => {
    const rating = shallow(<RatingWithNum rating={5.0}/>);
    expect(rating.exists()).toBeTruthy();
  });

  it("can handle a rating of 5 correctly", () => {
    const rating = shallow(<RatingWithNum rating={5.0}/>);
    // console.log(rating.debug({ verbose: true }));
    expect(rating.text()).toBe("<StarRating />5");
  });

  it("can handle a rating with a float value correctly", () => {
    const rating = shallow(<RatingWithNum rating={2.5}/>);
    // console.log(rating.debug({ verbose: true }));
    expect(rating.text()).toBe("<StarRating />2.5");
  });

  it("can handle a rating of 0 correctly (should not show decimal point and should instead show whole number", () => {
    const rating = shallow(<RatingWithNum rating={0.0}/>);
    // console.log(rating.debug({ verbose: true }));
    expect(rating.text()).toBe("<StarRating />0");
  });

  /* Snapshot tests - creates snapshots for new tests, otherwise will check the existing tests. To update existing tests,
  follow the prompts on the screen after running test. Done to avoid regression. */

  it("renders with minimal props", () => {
    const rating = renderer.create(<RatingWithNum rating={5.0}/>);
    expect(rating).toMatchSnapshot();
  });

  it("renders with double star rating", () => {
    const rating = renderer.create(<RatingWithNum rating={2.5}/>);
    expect(rating).toMatchSnapshot();
  });

  it("renders correctly when editable", () => {
    const rating = renderer.create(<RatingWithNum rating={0.0}/>);
    expect(rating).toMatchSnapshot();
  });
});
