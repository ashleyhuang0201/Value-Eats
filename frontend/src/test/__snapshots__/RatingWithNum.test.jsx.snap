// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingWithNum renders correctly when editable 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <span
      aria-label="0 Stars"
      className="MuiRating-root MuiRating-readOnly"
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      role="img"
    >
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "0%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "0%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "0%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "0%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "0%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
    </span>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
    style={
      Object {
        "marginLeft": "10px",
        "marginTop": "4px",
      }
    }
  >
    0
  </div>
</div>
`;

exports[`RatingWithNum renders with double star rating 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <span
      aria-label="2.5 Stars"
      className="MuiRating-root MuiRating-readOnly"
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      role="img"
    >
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "0%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "0%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "50%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "0%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "0%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
    </span>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
    style={
      Object {
        "marginLeft": "10px",
        "marginTop": "4px",
      }
    }
  >
    2.5
  </div>
</div>
`;

exports[`RatingWithNum renders with minimal props 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <span
      aria-label="5 Stars"
      className="MuiRating-root MuiRating-readOnly"
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      role="img"
    >
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "0%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "0%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "0%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "0%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        className="MuiRating-decimal"
      >
        <span
          style={
            Object {
              "overflow": "hidden",
              "position": "absolute",
              "width": "0%",
              "zIndex": 1,
            }
          }
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span
          style={Object {}}
        >
          <span
            className="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
    </span>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
    style={
      Object {
        "marginLeft": "10px",
        "marginTop": "4px",
      }
    }
  >
    5
  </div>
</div>
`;
